### DISCUSSION TITLE
🚀💫🛸 The Future No Longer Belongs to a Different Loop!

### DISCUSSION THREAD

**Discussion by DavidDragonsage** [lllyasviel/Fooocus#3369] [General]
For as long as I can remember, every second time Fooocus started, this Console message would occur:
"The future belongs to a different loop than the one specified as the loop argument"

This message would occur after "UNet ADM Dimension 3816". When I saw the message, I knew that if I waited for the loading sequence to continue (which may or may not happen) Fooocus would in any event be unable to generate anything. The fix was to close the console and restart Fooocus, then all would be OK.

This Python generated message, that makes me think of time loops in Sci-Fi scenarios, seemed to be related to my inadequate hardware not being capable of meeting asynchronous timing deadlines (4 GB VRAM, but an old and unsupported Nvidia card). My wife's computer, with much more advanced hardware, never exhibited this problem. For that reason I never reported the time loops here.

But I am surprised and happy to report that since the release of Fooocus 2.50, this problem has never returned! 😁 

@mashb1t - Thank you for all your hard work, really appreciate it!

**Comment by mashb1t** [comment]
Thank you for the feedback, glad the code rework of the async_worker helped to fix this issue while also making the code more flexible by extracting the single 800 lines method into multiple others, laying the groundworks for the enhance feature.
